sys/net/link_layer/l2filter/l2filter.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_l2filter
+ * @{
+ *
+ * @file
+ * @brief       Link layer address filter implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "assert.h"
+#include "net/l2filter.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+static inline bool match(const l2filter_t *filter,
+                         const void *addr, size_t addr_len)
+{
+    return ((filter->addr_len == addr_len) &&
+            (memcmp(filter->addr, addr, addr_len) == 0));
+}
+
+void l2filter_init(l2filter_t *list)
+{
+    assert(list);
+
+    for (unsigned i = 0; i < L2FILTER_LISTSIZE; i++) {
+        list[i].addr_len = 0;
+    }
+}
+
+int l2filter_add(l2filter_t *list, const void *addr, size_t addr_len)
```

Comment:
I decided for (1) intentionally. IMHO this feature is basically made for testing and experiments, where in both cases one has pretty good control and knowledge what one is doing. So there is IMHO no need to catch duplicate entries...